contracts/governance/Staking/Staking.sol

```diff
@@ -90,32 +93,78 @@ contract Staking is WeightedStaking{
      * @param amount the amount of SOV tokens
      * @param stakeFor the address for which we want to increase the stake. staking for the sender if 0x0
      * @param until the lock date until which the funds are staked
-     */
-    function increaseStake(uint96 amount, address stakeFor, uint until) public{
+     * */
+    //TODO deprecated, can be removed when frontend updated
+    function increaseStake(uint96 amount, address stakeFor, uint until) public {
         require(amount > 0, "Staking::increaseStake: amount of tokens to stake needs to be bigger than 0");
+
         until = timestampToLockDate(until);
         uint96 balance = currentBalance(stakeFor, until);
         require(balance > 0, "Staking:increaseStake: nothing staked yet until the given date. Use 'stake' instead.");
-        
+
+        //stake for the msg.sender if not specified otherwise
+        if(stakeFor == address(0)) {
+            stakeFor = msg.sender;
+        }
+
+        _increaseStake(amount, stakeFor, until);
+
+        _increaseDelegateStake(delegates[stakeFor][until], until, amount);
+    }
+
+    function _increaseStake(uint96 amount, address stakeFor, uint until) internal {
         //retrieve the SOV tokens
         bool success = SOVToken.transferFrom(msg.sender, address(this), amount);
         require(success);
-        
-        //stake for the msg.sender if not specified otherwise
-        if(stakeFor == address(0))
-            stakeFor = msg.sender;
-        
+
         //increase staked balance
+        uint96 balance = currentBalance(stakeFor, until);
         balance = add96(balance, amount, "Staking::increaseStake: balance overflow");
-        
+
         //update checkpoints
         _increaseDailyStake(until, amount);
-        _increaseDelegateStake(delegates[stakeFor][until], until, amount);
         _increaseUserStake(stakeFor, until, amount);
-        
+
         emit TokensStaked(stakeFor, amount, until, balance);
     }
-    
+
+    /**
+     * @notice stakes tokens according to the vesting schedule
+     * @param amount the amount of tokens to stake
+     * @param cliff the time interval to the first withdraw
+     * @param duration the staking duration
+     * @param intervalLength the length of each staking interval when cliff passed
+     * @param stakeFor the address to stake the tokens for or 0x0 if staking for oneself
+     * @param delegatee the address of the delegatee or 0x0 if there is none.
+     * */
+    function stakeTokens(
```

Comment:
maybe we can rename this function, so it's clearer what it does? like stakeBySchedule? or stakeVested?

Reply:
renamed to stakeBySchedule